src/Files.App/Helpers/AdaptiveLayoutHelpers.cs

```diff
@@ -3,124 +3,126 @@
 using Files.App.ViewModels;
 using Files.App.ViewModels.Previews;
 using Files.Backend.Services.Settings;
+using IniParser.Model;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using Windows.Storage;
+using static Files.App.Constants.AdaptiveLayout;
+using IO = System.IO;
 
 namespace Files.App.Helpers
 {
 	public static class AdaptiveLayoutHelpers
 	{
-		public static bool PredictLayoutMode(FolderSettingsViewModel folderSettings, string path, IList<ListedItem> filesAndFolders)
+		private static readonly IFoldersSettingsService foldersSettingsService = Ioc.Default.GetRequiredService<IFoldersSettingsService>();
+
+		public static void ApplyAdaptativeLayout(FolderSettingsViewModel folderSettings, string path, IList<ListedItem> filesAndFolders)
 		{
-			IUserSettingsService userSettingsService = Ioc.Default.GetRequiredService<IUserSettingsService>();
+			if (!foldersSettingsService.EnableOverridingFolderPreferences)
+				return;
+			if (string.IsNullOrWhiteSpace(path))
+				return;
+			if (folderSettings.IsLayoutModeFixed || !folderSettings.IsAdaptiveLayoutEnabled)
+				return;
```

Comment:
I think it would be more efficient to have only one `if` with many `ORs`

Reply:
I think it's okay like this, makes it more readable.